File: LeetCode75/Validate_Binary_Search_Tree/Validate_Binary_Search_Tree.cc
// 98. Validate Binary Search Tree C++
// Definition for a binary tree node.
#include <vector>

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class Solution {
public:
    std::vector<int> output;
    bool isValidBST(TreeNode* root) {
        if (!root) return true;
        nextNode(root);
        int pre = output[0];
        for (int i = 1; i < output.size(); i++) {
            if (output.at(i) <= pre) return false;
            pre = output.at(i);
        }
        return true;
    }
    void nextNode(TreeNode* root) {
        if (!root) return;
        nextNode(root->left);
        output.emplace_back(root->val);
        nextNode(root->right);
    }
};

File: LeetCode75_LevelOne/Validate_Binary_Search_Tree/Validate_Binary_Search_Tree.py
# 98. Validate Binary Search Tree Python
# Definition for a binary tree node.
class TreeNode(object):
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution(object):
    def isValidBST(self, root):
        """
        :type root: TreeNode
        :rtype: bool
        """
        output = []
        self.nextNode(root, output)

        for i in range(1, len(output)):
            if output[i-1] >= output[i]: return False
        return True

    def nextNode(self, root, output):
        if not root: return
        self.nextNode(root.left, output)
        output.append(root.val)
        self.nextNode(root.right, output)